python/hail/methods/statgen.py

```diff
@@ -1466,46 +1465,60 @@ def pca(entry_expr, k=10, compute_loadings=False, as_array=False):
     loadings = from_option(r._3())
     if loadings:
         loadings = Table(loadings)
-    return (jiterable_to_list(r._1()), scores, loadings)
+    return jiterable_to_list(r._1()), scores, loadings
 
 
-@typecheck(dataset=MatrixTable,
-           k=int,
+@typecheck(mt=MatrixTable,
            maf=numeric,
-           block_size=int,
+           k=nullable(int),
+           scores=nullable(Table),
            min_kinship=numeric,
-           statistics=enumeration("phi", "phik2", "phik2k0", "all"))
-def pc_relate(dataset, k, maf, block_size=512, min_kinship=-float("inf"), statistics="all"):
-    """Compute relatedness estimates between individuals using a variant
-    of the PC-Relate method.
+           statistics=enumeration('phi', 'phik2', 'phik2k0', 'all'),
+           block_size=nullable(int))
+def pc_relate(mt, maf, k=None, scores=None, min_kinship=-float("inf"), statistics="all", block_size=None):
+    """Compute relatedness estimates between individuals using a variant of the
+    PC-Relate method.
 
     .. include:: ../_templates/experimental.rst
 
     .. include:: ../_templates/req_tvariant.rst
 
     .. include:: ../_templates/req_biallelic.rst
 
+    .. include:: ../_templates/req_tstring.rst
```

Comment:
this should not require strings.

Reply:
I thought I re-wrote this to avoid strings, is there somewhere I forgot to do that?

Reply:
The generalization was causing problems, so I pulled it back to get to something that worked. I'll check again if it's simple to re-generalize (I put FIXME), trying to keep this PR to a merge-able improvement that includes Konrad's use case.